code/modules/mob/living/simple_animal/friendly/corgi.dm

```diff
@@ -75,7 +78,23 @@
 	..()
 	regenerate_icons()
 
-/mob/living/simple_animal/pet/corgi/sac_act(var/obj/effect/rune/R, victim)
+/mob/living/simple_animal/pet/corgi/attackby(var/obj/item/O as obj, var/mob/user as mob)
+	if(user.stat == CONSCIOUS && istype(O, /obj/item/weapon/razor))
+		if (shaved)
+			user << "<span class='warning'>You can't shave this corgi, it's already been shaved.</span>"
+			return
+		user.visible_message("<span class='notice'>[user] starts to shave [src] using \the [O].</span>")
+		if(do_after(user, 50))
+			user.visible_message("<span class='notice'>[user] shaves [src]'s hair using \the [O]. </span>")
+			playsound(loc, 'sound/items/Welder2.ogg', 20, 1)
+			shaved = 1
+			icon_state = "[shaved_state]"
+			icon_living = "[shaved_state]"
+			icon_dead = "[shaved_state]_dead"

```

Comment:
You could use `icon_state = "[icon_living]_shaved"` `icon_living = "[icon_living]_shaved"` `icon_dead = "[icon_living]_shaved_dead"` so that you don't have to add an shaved_state var.